### ISSUE TITLE
Downloading images causes ANR

### ISSUE THREAD

**Issue by msfjarvis** [Tunous/Dank#51]
**Describe the bug**
Trying to download a image from a Reddit post causes an ANR.

**To Reproduce**
1. Go to any post with a picture
2. Click on the picture
3. Press the download button
4. A notification is shown saying 'Queued 'https://i.redd.it/...' but the download never completes.

**Expected behavior**
The image downloads fully and a media notification is posted for it.

**Screenshots**
![](https://i.imgur.com/5rq2YHk.jpg)

**Smartphone (please complete the following information):**
 - Device: Redmi Note 5 Pro
 - OS: AOSP 9.0
 - Version 0.7.3 (6115aeb9ac8f11189f414c32cb6a2d6ca87f549e)

**Additional context**
I managed to extract the ANR trace, from my device: https://del.dog/xokoziyeje.pl

**Comment by msfjarvis** [comment]
Both debug and release builds exhibit the same behaviour, FWIW.

**Comment by Esselans** [comment]
> Both debug and release builds exhibit the same behaviour, FWIW.

Confirmed on master. On my 2 PR prior #44 downloads are working as expected.

**Comment by Tunous** [comment]
I've pushed ce2d1b83a5de8f943e9ffedad6461e9a91a2de27 which temporarily fixes this issue by downgrading Glide to version 4.8.0. The problem lies somewhere in changes in version 4.9.0 of Glide.

I did not yet manage to find the actual problem so this issue should stay open until someone does that.

**Comment by saket** [comment]
Wanna submit a bug report to Glide?

**Comment by Tunous** [comment]
Yes probably, but first need to find a real cause and create isolated sample if possible.
